staging/src/k8s.io/apiserver/pkg/util/flag/globalflags.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+
+	// libs that provide registration functions
+	"k8s.io/apiserver/pkg/util/logs"
+
+)
+
+// AddGlobalFlags explicitly registers flags that libraries (glog, verflag, etc.) register
+// against the global flagsets from "flag" and "github.com/spf13/pflag".
+// We do this in order to prevent unwanted flags from leaking into the kubernetes component's flagset.
+func AddGlobalFlags(fs *pflag.FlagSet) {
```

Comment:
At the moment, this function only registers flags from `glog` and internal libraries like `k8s.io/apiserver/pkg/util/logs`. I'm not too sure if moving the `verflag` back here would be a better idea since it is still referring to a global flag and most (I think all?) commands would need it.

Reply:
move the `verflag.go` need more disscuss